Week 04/Lecture 08/Assignment 3/src/main/java/com/lecture8/assignment3/repository/EmployeeRepository.java

```diff
@@ -0,0 +1,61 @@
+package com.lecture8.assignment3.repository;
+
+import java.util.List;
+
+import javax.sql.DataSource;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.IncorrectResultSizeDataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import com.lecture8.assignment3.entity.Employee;
+
+@Repository
+public class EmployeeRepository {
```

Comment:
Try to handle exception here

Reply:
I have update the code, All methods in EmployeeRepository now include try-catch blocks to handle potential exceptions. The code logs exceptions and returns appropriate failure codes or empty results in case of errors..